### PR TITLE
Fixed initial warnings caused by invalid propTypes and array indices

### PR DESCRIPTION
**Pull request by amatyas001**
Hey there!
I fixed some propTypes definition. Not all of them, it's still work in progress. Also fixed array map functions using indices as keys. Read more: [Why using an index as Key in React is probably a bad idea?](https://medium.com/@vraa/why-using-an-index-as-key-in-react-is-probably-a-bad-idea-7543de68b17c).

### CODE REVIEW

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -303,15 +309,15 @@ const Header = ({
                       <ListItemText primary={strings.app_my_profile} />
                     </ListItem>
                   </DrawerLink>
-                  <DrawerLink as="a" href={`${process.env.REACT_APP_API_HOST}/logout`}>
+                  <DrawerLink as="a" to={`${process.env.REACT_APP_API_HOST}/logout`}>
```
Comment (howardchung) [inline src/components/Header/Header.jsx]: I'm pretty sure this needs to be href since the logout is not part of the app

File: src/components/Header/Header.jsx [hunk 2]
```diff
@@ -109,24 +109,30 @@ const LinkGroup = ({ navbarPages }) => (
...
 const SettingsGroup = ({ children }) => {
   const [anchorEl, setAnchorEl] = useState(null);
   const handleClose = useCallback(() => {
-    setAnchorEl(undefined);
+    setAnchorEl(null);
   }, [setAnchorEl]);
 
   return (
-    <>
+    <div>
```
Comment (howardchung) [inline src/components/Header/Header.jsx]: what's the purpose of this change?

File: src/index.js [hunk 3]
```diff
@@ -40,7 +40,7 @@ const app = (
   </Provider>
 );
 
-if (rootElement.hasChildNodes()) {
+if (!rootElement.hasChildNodes()) {
```
Comment (howardchung) [inline src/index.js]: why this change?

File: src/components/Header/Header.jsx (hunk 2 above)
Comment (amatyas001) [reply src/components/Header/Header.jsx]: It's related to the ref forwarding in MUI. It doesn't work with fragments but should be with divs. It's still drops warning so... just a try.

File: src/components/Header/Header.jsx (hunk 1 above)
Comment (amatyas001) [reply src/components/Header/Header.jsx]: It works well with the login button in local env.  Thought it will work here too.

File: src/index.js (hunk 3 above)
Comment (amatyas001) [reply src/index.js]: It throws a warning about expected divs. Is SSR enabled here or why the hydrate?
